packages/contracts/src/dollar/UbiquityChef.sol

```diff
@@ -253,47 +286,53 @@ contract UbiquityChef is ReentrancyGuard {
         }
 
         if (isPriceDiffEnough) {
-            governanceMultiplier = IUbiquityFormulas(manager.formulasAddress())
-                .governanceMultiply(governanceMultiplier, currentPrice);
-            lastPrice = currentPrice;
+            governanceMultiplier_ =
+                _governanceMultiply(governanceMultiplier_, currentPrice);
+            lastPrice_ = currentPrice;
         }
+        lastPrice = lastPrice_;
+        governanceMultiplier = governanceMultiplier_;
     }
 
     // Update reward variables of the given pool to be up-to-date.
     function _updatePool() internal {
+        _updateGovernanceMultiplier();
         if (block.number <= pool.lastRewardBlock) {
             return;
         }
-        _updateGovernanceMultiplier();
 
         if (_totalShares == 0) {
             pool.lastRewardBlock = block.number;
             return;
         }
         uint256 multiplier = _getMultiplier();
-        uint256 governanceReward = (multiplier * governancePerBlock) / 1e18;
+        uint256 governanceReward = (multiplier * governancePerBlock) / 1e6;
+        pool.accGovernancePerShare =
+            pool.accGovernancePerShare + (governanceReward / _totalShares);
+        pool.lastRewardBlock = block.number;
+
+        pool.accGovernancePerShare = pool.accGovernancePerShare
+            + ((multiplier * governancePerBlock) / _totalShares / 1e6);
+        pool.lastRewardBlock = block.number;
         IERC20Ubiquity(manager.governanceTokenAddress()).mint(
             address(this), governanceReward
         );
         // mint another x% for the treasury
         IERC20Ubiquity(manager.governanceTokenAddress()).mint(
             manager.treasuryAddress(), governanceReward / governanceDivider
         );
-        pool.accGovernancePerShare = pool.accGovernancePerShare
-            + ((governanceReward * 1e12) / _totalShares);
-        pool.lastRewardBlock = block.number;
     }
 
-    // Safe Governance Token transfer function, just in case if rounding
-    // error causes pool to not have enough Governance Tokens.
+    // Safe governance transfer function, just in case if rounding
+    // error causes pool to not have enough governance tokens.
     function _safeGovernanceTransfer(address _to, uint256 _amount) internal {
-        IERC20Ubiquity governanceToken =
+        IERC20Ubiquity governance =
             IERC20Ubiquity(manager.governanceTokenAddress());
-        uint256 governanceBalance = governanceToken.balanceOf(address(this));
-        if (_amount > governanceBalance) {
-            governanceToken.safeTransfer(_to, governanceBalance);
+        uint256 governanceBal = governance.balanceOf(address(this));
+        if (_amount > governanceBal) {
+            governance.safeTransfer(_to, governanceBal);
```

Comment:
```suggestion
            governance.safeTransfer(_to, governanceBalance);
```

We should strive to never go shorthand in order to maximize codebase readability.

Reply:
Changed to governanceBalance